server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreMetadataLockManager.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.lockmanager;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.opensearch.index.store.RemoteDirectory;
+
+import java.io.IOException;
+import java.util.Collection;
+
+/**
+ * A Class that implements Remote Store Lock Manager by creating lock files.
+ * @opensearch.internal
+ */
+public class RemoteStoreMetadataLockManager implements RemoteStoreLockManager {
+    private static final Logger logger = LogManager.getLogger(RemoteStoreMetadataLockManager.class);
+    private final RemoteDirectory lockDirectory;
+    public RemoteStoreMetadataLockManager(RemoteDirectory lockDirectory) {
+        this.lockDirectory = lockDirectory;
+    }
+
+    @Override
+    public void acquire(LockInfo lockInfo) throws IOException {
+        assert lockInfo instanceof ShardLockInfo : "lockinfo should be instance of ShardLockInfo";
+        assert ((ShardLockInfo) lockInfo).getFileToLock() != null : "file to be locked should be provided while " +
+            "acquiring lock";
+        assert ((ShardLockInfo) lockInfo).getResourceId() != null : "resource Id should be provided while " +
+            "acquiring lock";
+        assert ((ShardLockInfo) lockInfo).getExpiryTime() != null : "expiry time for lock should be provided while " +
+            "acquiring lock";
```

Comment:
These checks should happen in `ShardLockInfo` class which is enforcing non nullability of these members as part of `getLockName()`. Also it should not be an assert as that will be disabled in test